cmd/analyzer/parse_args.go

```diff
@@ -180,9 +205,74 @@ func ParseInArgs(cmdlineArgs []string) (*InArgs, error) {
 	if err != nil {
 		return nil, err
 	}
+	err = invalidArgsQueryMode(&args, flagset)
+	if err != nil {
+		return nil, err
+	}
 
 	return &args, nil
 }
+
+func isFlagPassed(name string, flagset *flag.FlagSet) bool {
+	found := false
+	flagset.Visit(func(f *flag.Flag) {
+		if f.Name == name {
+			found = true
+		}
+	})
+	return found
+}
+
+func isQueryModeParamsPassed(flagset *flag.FlagSet) bool {
+	if isFlagPassed(QProtocol, flagset) || isFlagPassed(QSrcMinPort, flagset) || isFlagPassed(QSrcMaxPort, flagset) ||
+		isFlagPassed(QDstMinPort, flagset) || isFlagPassed(QDstMaxPort, flagset) {
+		return true
+	}
+
+	return false
+}
+
+func validRangeConnectionQueryMode(args *InArgs) error {
+	if *args.QSrcMinPort > *args.QSrcMaxPort {
+		return fmt.Errorf("srcMaxPort %d should be higher than srcMinPort %d", *args.QSrcMaxPort, *args.QSrcMinPort)
+	}
+
+	if *args.QSrcMinPort > common.MaxPort || *args.QSrcMinPort < common.MinPort ||
+		*args.QSrcMaxPort > common.MaxPort || *args.QSrcMaxPort < common.MinPort {
+		return fmt.Errorf("srcMaxPort and srcMinPort must be in ranges [%d, %d]", common.MinPort, common.MaxPort)
+	}
+
+	if *args.QDstMinPort > *args.QDstMaxPort {
+		return fmt.Errorf("DstMaxPort %d should be higher than DstMinPort %d", *args.QSrcMaxPort, *args.QSrcMinPort)
+	}
+
+	if *args.QDstMinPort > common.MaxPort || *args.QDstMinPort < common.MinPort ||
+		*args.QDstMaxPort > common.MaxPort || *args.QDstMaxPort < common.MinPort {
+		return fmt.Errorf("DstMaxPort and DstMinPort must be in ranges [%d, %d]", common.MinPort, common.MaxPort)
+	}
+
```

Comment:
Please consider having the checks for `dst/src` in a function called once, to avoid code duplication